influence_benchmark/RL/base_iteration.py

```diff
@@ -146,39 +144,54 @@ def _save_kwargs(self, kwargs):
         with open(str(self.traj_dir / "kwargs.yaml"), "w+") as outfile:
             yaml.dump(self.kwargs_to_save, outfile, default_flow_style=False)
 
+    def setup_backends(self, agent_device, env_device, lora_path=None):
+        backends = {}
+        if agent_device != env_device or self.inference_quantization is not None:
+            # Assuming that if this is the case, we can't share any backend at all. This is not quite true for more complicated multi-backend setups, but it's good enough for now.
+            for model_type, model_name in self.model_names.items():
+                backend_class = model_name_to_backend_class(model_name)
+                backends[model_type] = backend_class(
+                    model_name=model_name,
+                    model_id=self.agent_model_id,  # type: ignore
+                    device=agent_device if model_type == "agent" else env_device,
+                    lora_path=lora_path,
```

Comment:
We probably want lora_path=None here unless  model_type == "agent". If we have different models this will crash because you can't e.g. load a gemma lora for a llama model. We don't want to use a lora for the env stuff

Reply:
The problem with this is that if both models are e.g. llamas 8B, we don't want to load the backend twice. But if we had one backend with lora_path=None and the other with a lora_path, we would have to load it twice no matter what?